src/Microsoft.OData.Edm/Csdl/CsdlXmlWriter.cs

```diff
@@ -96,22 +127,50 @@ private void WriteEFCsdl()
             this.writer.Flush();
         }
 
+        private async Task WriteEFCsdlAsync()
+        {
+            await this.WriteEdmxElementAsync();
+            await this.WriteRuntimeElementAsync();
+            await this.WriteConceptualModelsElementAsync();
+            await this.WriteSchemasAsync();
+            await this.EndElementAsync(); // </ConceptualModels>
+            await this.EndElementAsync(); // </Runtime>
+            await this.EndElementAsync(); // </Edmx>
+            await this.writer.FlushAsync();
+        }
+
         private void WriteEdmxElement()
         {
             this.writer.WriteStartElement(CsdlConstants.Prefix_Edmx, CsdlConstants.Element_Edmx, this.edmxNamespace);
             this.writer.WriteAttributeString(CsdlConstants.Attribute_Version, GetVersionString(this.edmxVersion));
         }
 
+        private async Task WriteEdmxElementAsync()
+        {
+            await this.writer.WriteStartElementAsync(CsdlConstants.Prefix_Edmx, CsdlConstants.Element_Edmx, this.edmxNamespace);
+            await this.writer.WriteAttributeStringAsync(null, CsdlConstants.Attribute_Version, null, GetVersionString(this.edmxVersion));
+        }
+
         private void WriteRuntimeElement()
         {
             this.writer.WriteStartElement(CsdlConstants.Prefix_Edmx, CsdlConstants.Element_Runtime, this.edmxNamespace);
         }
 
+        private async Task WriteRuntimeElementAsync()
+        {
+            await this.writer.WriteStartElementAsync(CsdlConstants.Prefix_Edmx, CsdlConstants.Element_Runtime, this.edmxNamespace);
+        }

```

Comment:
In cases where you're only performing one async call in the method, you could just "foward" the returned task instead of using `async/await` this way we can avoid the overhead of the state machine associated with `async/await`.
```suggestion
        private Task WriteRuntimeElementAsync()
        {
            return this.writer.WriteStartElementAsync(CsdlConstants.Prefix_Edmx, CsdlConstants.Element_Runtime, this.edmxNamespace);
        }

```

Reply:
The reason why I used `async/await` in this case is to avoid unobserved task exceptions and unexpected behavior if the caller expects the operation to be completed when the method returns.

Reply:
@habbes I have resolve this. The reason why I used `async/await` in this case was to avoid unobserved task exceptions and unexpected behavior. `Non-awaited Tasks hide exceptions.`

Reply:
@WanjohiSammy this is a well-known performance improvement technique wrt async - it will not have result in any unexpected behaviour. Search/read about "eliding async/await "